src/pages/[platform]/build-a-backend/functions/examples/create-user-profile-record/index.mdx

```diff
@@ -37,41 +37,81 @@ You can use `defineAuth` and `defineFunction` to create a [Cognito post confirma
 A user is "confirmed" when they verify their account. Typically this happens when the user confirms their email via the verification email. The post confirmation handler will _not_ be triggered for federated sign-ins (i.e. social sign-in).
 
 </Callout>
-
 To get started, install the `aws-lambda` package, which is used to define the handler type.
 
 ```bash title="Terminal" showLineNumbers={false}
 npm add --save-dev @types/aws-lambda
 ```
 
-Next, create a new directory and a resource file, `amplify/auth/post-confirmation/resource.ts`. Then, define the Function with `defineFunction`:
+Update the `amplify/data/resource.ts` file to define a data model for the user's profile:
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
+import { postConfirmation } from "../auth/post-confirmation/resource";
+
+const schema = a
+  .schema({
+    UserProfile: a
+      .model({
+        email: a.string(),
+        profileOwner: a.string(),
+      })
+      .authorization((allow) => [
+        allow.ownerDefinedIn("profileOwner"),
+      ]),
+  })
+  .authorization((allow) => [allow.resource(postConfirmation)]);
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: "apiKey",
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+
+```
+
+
+Create a new directory and a resource file, `amplify/auth/post-confirmation/resource.ts`. Then, define the Function with `defineFunction`:
 
 ```ts title="amplify/auth/post-confirmation/resource.ts"
 import { defineFunction } from '@aws-amplify/backend';
 
 export const postConfirmation = defineFunction({
-  name: 'postConfirmation',
+  name: 'post-confirmation',
 });
 ```
 
-Then, create the corresponding handler file, `amplify/auth/post-confirmation/post-confirmation/handler.ts`, file with the following contents:
+Run the command `npx amplify sandbox` to create the backend, then use the command below to generate GraphQL client code to call your data backend.
+<Callout info>
+**Note** We are working on bringing the end-to-end typed experience to connect to your data from within function resources without needing this step. If you'd like to provide feedback on the experience or want to have early access, join our [Discord community](https://discord.gg/amplify).
+</Callout>
```

Comment:
Same comment as above:
- I think we need new lines between the `<Callout>`
- Should we have a colon after `Note`?

```suggestion
<Callout info>

**Note**: We are working on bringing the end-to-end typed experience to connect to your data from within function resources without needing this step. If you'd like to provide feedback on the experience or want to have early access, join our [Discord community](https://discord.gg/amplify).

</Callout>
```